### PR TITLE
Show a notification dot on toggle menu button in narrow mode

### PR DESCRIPTION
**Pull request by balloob**
- When in narrow mode, show a dot on the toggle menu button to indicate that there are notifications.
- When not in narrow mode:
  - hide the menu button on the page
  - show menu button in sidebar

image

### CODE REVIEW

File: src/components/ha-menu-button.ts [hunk 1]
```diff
@@ -5,34 +5,113 @@ import {
...
-  protected createRenderRoot(): Element | ShadowRoot {
-    return this;
+  protected updated(changedProps) {
+    super.updated(changedProps);
+
+    if (!changedProps.has("narrow")) {
+      return;
+    }
+
+    this.style.visibility = this.narrow ? "initial" : "hidden";
```
Comment (bramkragten) [inline src/components/ha-menu-button.ts]: Wouldn't `display: none` be better? Now you have a lot of margin in front of the header text when the button is hidden.

Comment (balloob) [reply src/components/ha-menu-button.ts]: It would misalign the text on Lovelace, I don't know what I dislike more.

Comment (bramkragten) [reply src/components/ha-menu-button.ts]: With the views menu?

Comment (balloob) [reply src/components/ha-menu-button.ts]: 60402023-81d4df00-9b3e-11e9-8d1b-e1b06249dd12

Comment (maykar) [reply src/components/ha-menu-button.ts]: What about adjusting the margin of both app-toolbar & paper-tabs on narrow?

Comment (balloob) [reply src/components/ha-menu-button.ts]: Which margin?

**Pull request merged**